### PR TITLE
AP_Scripting: add missing defines to build on Periph

### PR DESCRIPTION
**Pull request by IamPete1**
This work is sponsored by Spektreworks.

This adds defines such that scripting can run on periph with most everything else disabled, obviously we already run on vehicles with everything else enabled, I have not tested all possible combinations in between.

With everything else turned off the flash cost is around 100k but as you add back in more library's the extra bindings means the over head increases. Its only really practical with 512kB flash or more. To use you would have to enable romfs and embed the script into the firmware. 

I have tested a "hello world" script on a CubeBlack periph heavy build.

### CODE REVIEW

File: libraries/AP_Scripting/AP_Scripting_config.h [hunk 1]
```diff
@@ -3,5 +3,15 @@
...
-#define AP_SCRIPTING_ENABLED BOARD_FLASH_SIZE > 1024
+#define AP_SCRIPTING_ENABLED (BOARD_FLASH_SIZE > 1024)
+#endif
+
+#if AP_SCRIPTING_ENABLED
+    #include <AP_Filesystem/AP_Filesystem_config.h>
+    #if !AP_FILESYSTEM_SYS_ENABLED
+        #error "Scripting requires a filesystem"
+    #endif
+    #if !(AP_FILESYSTEM_FATFS_ENABLED || AP_FILESYSTEM_ROMFS_ENABLED)
```
Comment (peterbarker) [inline libraries/AP_Scripting/AP_Scripting_config.h:14]:
```suggestion
    #if AP_FILESYSTEM_FILE_READING_ENABLED
```

File: libraries/AP_Scripting/AP_Scripting_config.h [hunk 2]
```diff
@@ -3,5 +3,15 @@
 #include <AP_HAL/AP_HAL_Boards.h>
 
 #ifndef AP_SCRIPTING_ENABLED
-#define AP_SCRIPTING_ENABLED BOARD_FLASH_SIZE > 1024
+#define AP_SCRIPTING_ENABLED (BOARD_FLASH_SIZE > 1024)
+#endif
+
+#if AP_SCRIPTING_ENABLED
+    #include <AP_Filesystem/AP_Filesystem_config.h>
+    #if !AP_FILESYSTEM_SYS_ENABLED
```
Comment (peterbarker) [inline libraries/AP_Scripting/AP_Scripting_config.h:11]: What's this one here for?

**Review by peterbarker — CHANGES_REQUESTED** [review]
I reviewed a single patch...

File: libraries/AP_Scripting/AP_Scripting_config.h (hunk 2 above)
Comment (IamPete1) [reply libraries/AP_Scripting/AP_Scripting_config.h]: I have no idea... fixed.

File: libraries/AP_Scripting/AP_Scripting_config.h (hunk 1 above)
Comment (IamPete1) [reply libraries/AP_Scripting/AP_Scripting_config.h]: done, thanks.

**Comment by rmackay9** [comment]
Looks great but maybe we need a simple autotest to be sure we get a warning when a new binding breaks this again?

**Review by tridge — APPROVED** [review]

**Pull request merged**